src/main/java/org/apache/cassandra/sidecar/restore/RestoreProcessor.java

```diff
@@ -78,6 +81,7 @@ public RestoreProcessor(ExecutorPools executorPools,
                             SSTableImporter importer,
                             RestoreSliceDatabaseAccessor sliceDatabaseAccessor,
                             RestoreJobUtil restoreJobUtil,
+                            CachedLocalTokenRanges localTokenRangesProvider,
```

Comment:
does this need to be the concrete implementation? Ideally we should just use the interface here: `org.apache.cassandra.sidecar.locator.LocalTokenRangesProvider`

Reply:
It is not required, but the current form of code is succinct, as `CachedLocalTokenRanges` is annotated with `@Singleton`. I would keep it for now, until it gets complex (unlikely).

Reply:
this wouldn't allow us to swap out to a new implementation and  it would strictly tie it to the cached implementation.  That's an anti pattern for dependency injection frameworks like guice